src/main/java/hudson/plugins/scm_sync_configuration/strategies/model/PatternsEntityMatcher.java

```diff
@@ -22,12 +27,33 @@ public boolean matches(Saveable saveable, File file) {
 		if (file == null) {
 			return false;
 		}
-		String filePathRelativeToHudsonRoot = JenkinsFilesHelper.buildPathRelativeToHudsonRoot(file);
-        AntPathMatcher matcher = new AntPathMatcher();
-        for(String pattern : includesPatterns) {
-            if(matcher.match(pattern, filePathRelativeToHudsonRoot)){
-                return true;
-            }
+		return matches(saveable, JenkinsFilesHelper.buildPathRelativeToHudsonRoot(file), file.isDirectory());
+	}
+
+	public boolean matches(Saveable saveable, String pathRelativeToRoot, boolean isDirectory) {
+		if (pathRelativeToRoot != null) {
+			// Guard our own SCM workspace and the war directory. User-defined includes might inadvertently include those if they start with * or **!
+			if (pathRelativeToRoot.equals(SCM_WORKING_DIRECTORY) || pathRelativeToRoot.startsWith(SCM_WORKING_DIRECTORY + '/')) {
+				return false;
+			} else if (pathRelativeToRoot.equals(WAR_DIRECTORY) || pathRelativeToRoot.startsWith(WAR_DIRECTORY + '/')) {
+				return false;
+			}
+	        AntPathMatcher matcher = new AntPathMatcher();
+	        String directoryName = null;
+	        for (String pattern : includesPatterns) {
+	            if (matcher.match(pattern, pathRelativeToRoot)) {
+	                return true;
+	            } else if (isDirectory) {
+	            	// pathRelativeFromRoot is be a directory, and the pattern end in a file name. In this case, we must claim a match.
```

Comment:
Can you drop an eye on this please: https://github.com/jenkinsci/scm-sync-configuration-plugin/pull/28/files#diff-8ea250ea1ccf94b3ce908b8029599565R26 

In case jobs directory is a link to a mounted volume so nextBuildNumber stays in sync with buildHistory and symlinks we don't want the scm-sync-configuration plugin to throw exceptions.

Such cases are usual when running Jenkins in Docker as a Phoenix Docker Container or simply when rehydrating the build history an keeping the nextBuildNumber in sync with rehydrated build history.